src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * The soft shrink function has threshold proportional to the noise level given
+ * by the user.
+ * The use of a Soft Shrink activation function provides adaptive denoising at
+ * various noise levels using a single CNN(Convolution Neural) without a
+ * requirement to train a unique CNN for each noise level.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * Soft Shrink operator is defined as,
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x - lambda & : x >  lambda \\
+ *     x + lambda & : x < -lambda \\
+ *     0 & : otherwise
+ *   \end{array} \\
+ * \right.
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x >  lambda \\
+ *     1 & : x < -lambda \\
+ *     0 & : otherwise
+ *   \end{array}
+ * \right.
+ * @f}
+ * lambda is set to 0.5 by default.
```

Comment:
Hi @favre49 
I am afraid I have included this there also, line no 61 - 63.
Also these are important here because here I have defined the formula.
Thanks.

Reply:
The reason I said this is because this seemed like an implementation detail that would belong with the constructor, not with the overview of the class, and putting it in both places would be redundant

Reply:
Hi @favre49 
I have implemented the changes.
Kindly see the same if they are satisfactory.
Thanks.